src/lib/system.ts

```diff
@@ -27,3 +27,17 @@ export const isDev = ENV.DEV;
 export const isProd = ENV.PROD;
 export const hasStripePublicKey = !!VARS.PUBLIC_STRIPE_KEY;
 export const GRACE_PERIOD_OVERRIDE = false;
+
+/**
+ * Checks if the endpoint belongs to the Appwrite Cloud domain family,
+ * including `cloud.appwrite.io` and region subdomains like `fra.cloud.appwrite.io`.
+ *
+ * Useful when building the API endpoint for custom domains `CNAME`'d to regional endpoints.
+ */
+export const isCloudHostname = (() => {
+    try {
+        return new URL(VARS.APPWRITE_ENDPOINT).hostname.endsWith('cloud.appwrite.io');
```

Comment:
I dont think hardcoding `cloud.appwrite.io` is good here. we should just check for console mode cloud

Reply:
It'll cause the same issue if you have the cloud repo running locally with console mode set to cloud.